lib/core/unzipomatic.ts

```diff
@@ -0,0 +1,31 @@
+import {Readable} from "stream";
+
+export type UnzipOptions = {
+    // ToDo, but I would start with making yauzl defaults non-configureable and see if there is demand for flexibility there
+}
+
+export type SourceType = string | File | Blob | Buffer | Readable
+
+export type TargetFileMetadata = {
+    fullPath: string
```

Comment:
is there any other useful metadata we could return?

Reply:
When this type will be used?

Reply:
Ok, but for perf reasons I think we can make this list creation optional.

Reply:
But in general, we can see at `entry` metadata to populate this type.